data/src/main/java/com/antosik/benchproject/data/movie/details/repo/MovieDetailsDataRepository.kt

```diff
@@ -1,12 +1,20 @@
 package com.antosik.benchproject.data.movie.details.repo
 
+import com.antosik.benchproject.data.movie.details.entity.toEntity
 import com.antosik.benchproject.data.movies.common.api.MoviesApi
+import com.antosik.benchproject.data.movies.common.database.MovieDao
 import com.antosik.benchproject.domain.movie.details.repo.MovieDetailsRepository
 
 internal class MovieDetailsDataRepository(
-    private val apiService: MoviesApi
+    private val apiService: MoviesApi,
+    private val dao: MovieDao
 ) : MovieDetailsRepository {
     override suspend fun getMovieDetails(movieId: Int) = runCatching {
-        apiService.getMovieDetails(movieId).toDomain()
+        apiService.getMovieDetails(movieId).let { movieDetailsRemote ->
+            dao.insertMovieDetails(movieDetailsRemote.toEntity())
+            dao.insertGenres(movieDetailsRemote.genres.toEntity(movieId))
+        }
+    }.run {
```

Comment:
Is it worth using another scope function? Maybe keep it all in `let`?

Reply:
No, I think it is not worth using so I change **run** to **let**